conversation/echo/echo.go

```diff
@@ -59,20 +59,31 @@ func (e *Echo) GetComponentMetadata() (metadataInfo metadata.MetadataMap) {
 	return
 }
 
-// Converse returns the last message's content directly.
+// Converse returns one output per input message.
 func (e *Echo) Converse(ctx context.Context, r *conversation.Request) (res *conversation.Response, err error) {
-	var content string
-	var toolCalls []llms.ToolCall
+	if r.Message == nil {
+		return &conversation.Response{
+			ConversationContext: r.ConversationContext,
+			Outputs:             []conversation.Result{},
+		}, nil
+	}
+
+	outputs := make([]conversation.Result, 0, len(*r.Message))
 
-	if r.Message != nil && len(*r.Message) > 0 {
-		lastMessage := (*r.Message)[len(*r.Message)-1]
+	for _, message := range *r.Message {
+		var content string
+		var toolCalls []llms.ToolCall
 
-		for _, part := range lastMessage.Parts {
+		for i, part := range message.Parts {
 			switch p := part.(type) {
 			case llms.TextContent:
+				// end with space if not the first part
```

Comment:
why we need this?

Reply:
this is keeping the same logic as before with alpha1 - good with swapping back to this to not break the alpha1 tests in dapr 👍🏻